app/src/pages/inside/dashboardItemPage/newWidgets/charts/launchesComparisonChart/launchesComparisonChart.jsx

```diff
@@ -0,0 +1,277 @@
+
+    return {
+      top: top - 8,
+      left,
+    };
+  };
+
+  getData = () => {
+    const { widget, intl, isPreview } = this.props;
+    const data = widget.content.result;
+    const chartData = {};
+    const chartDataOrdered = [];
+    const colors = {};
+    const contentFields = widget.content_parameters.content_fields;
+
+    this.itemData = [];
+
+    Object.keys(data[0].values).forEach((key) => {
+      chartData[key] = [key];
+      colors[key] = chartConfigs.colors[key.split('$')[2]];
+    });
+
+    data.forEach((item) => {
+      this.itemData.push({
+        id: item.id,
+        name: item.name,
+        number: item.number,
+        startTime: item.startTime,
+      });
+      Object.keys(item.values).forEach((key) => {
+        const val = item.values[key];
+        chartData[key].push(val);
+      });
+    });
+
+    contentFields.forEach((key) => {
+      if (key === 'statistics$executions$total') {
+        return;
+      }
+      chartDataOrdered.push(chartData[key]);
+    });
+
+    this.itemNames = chartDataOrdered.map((item) => item[0]);
+
+    this.config = {
+      data: {
+        columns: chartDataOrdered,
+        type: 'bar',
+        onclick: this.onChartClick,
+        order: null,
+        colors,
+      },
+      grid: {
+        y: {
+          show: !isPreview,
+        },
+      },
+      axis: {
+        x: {
+          show: !isPreview,
+          type: 'category',
+          categories: this.itemData.map((item) => `#${item.number}`),
+          tick: {
+            centered: true,
+            inner: true,
+            outer: false,
+          },
+          lines: [{ value: 5, text: 'Label', class: 'color-grid' }],
+        },
+        y: {
+          show: !isPreview,
+          padding: {
+            top: 0,
+          },
+          max: 100,
+          label: {
+            text: `% ${intl.formatMessage(messages.ofTestCases)}`,
+            position: 'outer-middle',
+          },
+        },
+      },
+      interaction: {
+        enabled: !isPreview,
+      },
+      padding: {
+        top: isPreview ? 0 : 85,
+        left: isPreview ? 0 : 60,
+        right: isPreview ? 0 : 20,
+        bottom: 0,
+      },
+      legend: {
+        show: false,
+      },
+      tooltip: {
+        grouped: false,
+        position: this.getPosition,
+        contents: this.contents,
+      },
+      size: {
+        height: this.props.height,
+      },
+    };
+  };
+
+  getCoords = ({ pageX, pageY }) => {
+    this.x = pageX;
+    this.y = pageY;
+  };
+
+  contents = (d, defaultTitleFormat, defaultValueFormat, color) => {
+    const { name, number, startTime } = this.itemData[d[0].index];
+    const id = d[0].id;
+    return renderTooltip({
+      launchName: name,
+      launchNumber: number,
+      startTime,
+      itemCases: d[0].value,
+      color: color(id),
+      itemName: this.props.intl.formatMessage(messages[id.split('$total')[0]]),
+    });
+  };
+
+  render() {
+    this.getData();
```

Comment:
Calling this function here looks very strange. Render method shouldn't have any side effects, so I think better to place this in other lifecycle methods.